lib/smack/SmackRep.cpp

```diff
@@ -1012,13 +1017,44 @@ std::string SmackRep::code(llvm::CallInst& ci) {
 
   std::string s = fmt;
   for (unsigned i=1; i<ci.getNumOperands()-1; i++) {
-    const Expr* a = arg(f, i, ci.getOperand(i));
+    Value* argV = ci.getOperand(i);
     std::string::size_type idx = s.find('@');
     assert(idx != std::string::npos && "inline code: too many arguments.");
 
+    llvm::Type* targetType = argV->getType();
+    bool isCast = false;
+    if (idx + 1 < s.length()) {
+      switch(s[idx+1]) {
+        case 'c':
```

Comment:
Does anyone think it would be bad to write this as:
```
case 'c':
case 'b':
case 'B':
```
LLVM doesn't distinguish between these types, but it might match up with the type in the surrounding C code.

Reply:
This is a good point. @shaobo-he any thoughts on this?